SQL_Assignment.txt

```diff
@@ -0,0 +1,1326 @@
+--->>SQL Assignment
+
+--->>Solved Queries
+
+1)	List all the columns of the Salespeople table.
+
+Ans :- mysql> SELECT * FROM SALESPEOPLE ;
++------+---------+-----------+------+
+| snum | sname   | city      | comm |
++------+---------+-----------+------+
+| 1001 | peel    | london    | 12   |
+| 1002 | seeres  | sanjoes   | 13   |
+| 1004 | motika  | london    | 11   |
+| 1003 | AlexRod | new york  | 10   |
+| 1007 | rifkin  | barcelona | 15   |
+| 1008 | fran    | london    | 25   |
++------+---------+-----------+------+
+6 rows in set (0.00 sec)
+
+2)	List all customers with a rating of 100.
+Ans:- mysql> SELECT * FROM costumers WHERE rating = 100 ;
++------+---------+--------+--------+-------+
+| cnum | cname   | city   | rating | sname |
++------+---------+--------+--------+-------+
+| 2001 | hoffman | london | 100    | 1001  |
+| 2006 | clemens | london | 100    | 1001  |
+| 2007 | pereira | rome   | 100    | 1004  |
++------+---------+--------+--------+-------+
+3 rows in set (0.00 sec)
+
+ 
+3)	Find the largest order taken by each salesperson on each date.
+
+Ans :- mysql> select salespeople.snum, salespeople.sname, max(amt) as MAX_ORDER from orders inner join salespeople on orders.snum=salespeople.snum group by snum;
```

Comment:
Grouping should be done on the OrderDate. This is a wrong query.

Reply:
mysql> select salespeople.snum, salespeople.sname, orders.odate, max(amt) as MAX_ORDER from orders inner join salespeople on orders.snum=salespeople.snum group by odate;
+------+---------+----------+-----------+
| snum | sname   | odate    | MAX_ORDER |
+------+---------+----------+-----------+
| 1007 | rifkin  | 10/03/90 | 767.19    |
| 1003 | AlexRod | 10/04/90 | 75.75     |
| 1001 | peel    | 10/05/90 | 4723.00   |
| 1002 | seeres  | 10/06/90 | 9891.88   |
+------+---------+----------+-----------+